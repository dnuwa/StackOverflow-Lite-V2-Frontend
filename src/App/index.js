import React from 'react';
import { Router, Route, Switch } from 'react-router-dom';
import Notification from 'react-notify-toast';
import Header from '../components/Header';
import '../styles/main.css';
import history from '../commons/history';
import loginContainer from '../containers/loginContainer';
import SiginUpContainer from '../containers/signupContainer';
import QuestionContainer from '../containers/questionsContainer';
import QuestionsContainer from '../containers/allQuestions';


const App = () => (
  <Router history={history}>
    <React.Fragment>
      <Header />
      <Notification />
      <Switch>
        <Route path="/" component={loginContainer} exact />
        <Route path="/login" component={loginContainer} />
        <Route path="/signup" component={SiginUpContainer} />
        <Route path="/questions" component={QuestionContainer} />
        <Route path="/allquestions" component={QuestionsContainer} />
      </Switch>
    </React.Fragment>
  </Router>
);

export default App;
