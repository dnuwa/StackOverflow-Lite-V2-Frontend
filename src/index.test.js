import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import App from './App';

describe('Provider and App', () => {
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const initialState = { question: {}, user: {} };
  const store = mockStore(initialState);
  const provider = shallow(<Provider store={store}><App /></Provider>);
  const app = shallow(<App />);

  it('renders <Provider/> correctly', () => {
    expect(provider).toMatchSnapshot();
    expect(app).toMatchSnapshot();
  });
});
