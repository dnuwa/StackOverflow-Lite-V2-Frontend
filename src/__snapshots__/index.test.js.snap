// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provider and App renders <Provider/> correctly 1`] = `<App />`;

exports[`Provider and App renders <Provider/> correctly 2`] = `
<BrowserRouter>
  <Header />
</BrowserRouter>
`;
